File: src/Offering/OfferingTest/UnitTests/LocalizationTest.cs
using Offering.Models;

namespace OfferingTest;

public class LocalizationTest
{
    [Fact]
    public void Localization_Is_Instantiated_Correctly()
    {
        var localization = Localization.Create("DE", "Germany", "Deutschland", "EUR");
        
        Assert.Equal("DE", localization.CountryCode);
        Assert.Equal("Germany", localization.CountryName);
        Assert.Equal("Deutschland", localization.LocalName);
        Assert.Equal("EUR", localization.Currency);
    }
}

File: src/OfferSB/src/test/java/com/example/offersb/test/UnitTests/LocalizationTest.java
package com.example.offersb.test.UnitTests;

import com.example.offersb.models.Localization;
import org.junit.jupiter.api.Test;

public class LocalizationTest {

    @Test
    public void testLocalization_Is_Instantiated_Correctly() {
        var localization = Localization.create("DE", "Germany", "Deutschland", "EUR");

        assert localization.getCountryCode() == "DE";
        assert localization.getCountryName() == "Germany";
        assert localization.getLocalName() == "Deutschland";
        assert localization.getCurrency() == "EUR";
    }
}
